src/server.rs

```diff
@@ -333,28 +352,40 @@ impl<State: Send + Sync + 'static> Server<State> {
 
         let mut incoming = listener.incoming();
         while let Some(stream) = incoming.next().await {
-            let stream = stream?;
-            let this = self.clone();
-            let local_addr = stream.local_addr().ok().map(|addr| format!("{:?}", addr));
-            let peer_addr = stream.peer_addr().ok().map(|addr| format!("{:?}", addr));
-            task::spawn(async move {
-                let res = async_h1::accept(stream, |mut req| async {
-                    req.set_local_addr(local_addr.as_ref());
-                    req.set_peer_addr(peer_addr.as_ref());
-                    let res = this.respond(req).await;
-                    let res = res.map_err(|_| io::Error::from(io::ErrorKind::Other))?;
-                    Ok(res)
-                })
-                .await;
-
-                if let Err(err) = res {
-                    log::error!("async-h1 error", { error: err.to_string() });
+            let stream = match stream {
+                Err(ref e) if is_transient_error(e) => continue,
+                Err(error) => {
+                    let delay = std::time::Duration::from_millis(500);
```

Comment:
This seems kind of large? I guess it should be pretty unusual so maybe it's fine. Ideally we'd have a dynamically increasing backoff here.

Reply:
I didn't have a good sense of an appropriate number here so I used the same number in the book section cited above

Reply:
Well, it's 500ms in the book because it's reasonably fine to write a log message every 500ms. Skipping log message always will make the issue a nightmare to debug. Sleeping less but keeping message rarely will substantially complicate the code (which is fine for prod, but bad for a book).

But at the end of the day, you should not expect code to hit this timeout for a long time. I always consider this as an infrastructure issue: either bump file descriptor limit if you have enough memory, or lower listen queue, or put a load-balancer/queue in front of the process to get rid of error. So large timeout here is fine. Also, TCP delays can be larger than 500ms, so any client should be fine to wait that much.